test/integration/util/util.go

```diff
@@ -407,21 +438,27 @@ func InitTestSchedulerWithOptions(
 		testCtx.InformerFactory,
 		testCtx.DynInformerFactory,
 		profile.NewRecorderFactory(eventBroadcaster),
-		testCtx.Ctx.Done(),
+		ctx.Done(),
 		opts...,
 	)
 
 	if err != nil {
 		t.Fatalf("Couldn't create scheduler: %v", err)
 	}
 
-	eventBroadcaster.StartRecordingToSink(testCtx.Ctx.Done())
+	eventBroadcaster.StartRecordingToSink(ctx.Done())
 
 	oldCloseFn := testCtx.CloseFn
 	testCtx.CloseFn = func() {
 		oldCloseFn()
 		eventBroadcaster.Shutdown()
 	}
+
+	testCtx.SchedulerCloseFn = func() {
+		cancel()
+		eventBroadcaster.Shutdown()
```

Comment:
We add `eventBroadcaster.Shutdown()` both here and above because sometimes we'll not call `SchedulerCloseFn` in cleanup, then eventBroadcaster may not be closed properly.